src/index.js

```diff
@@ -104,7 +104,9 @@ export default class FeatureGates {
           return acc;
         }, {});
     }
-    return this.featureGates;
+
+    // deep-copy the response object to prevent mutability to original object in use
+    return JSON.parse(JSON.stringify(this.featureGates));
```

Comment:
Just thinking out loud. Are there any possibilities for cyclic dependencies here? Is `this.featureGates` always a plain JSON object

Reply:
Also add some test cases to verify the condition that caused this change @abhijith-vijayan